tree/tree/src/TBasket.cxx

```diff
@@ -1284,8 +1301,16 @@ Int_t TBasket::WriteBuffer()
             R__SizeBuffer(*cascade_buffer, buflen);
             cascade_buffer->SetWriteMode();
             precisionCascades.push_back( cascade_buffer->Buffer() + cascade_basket->GetKeylen() );
+            cxlevels.push_back(cxlevel); // WRONG! From where do we get the cxlevels of the precision cascade?
          }
       }
+      auto doPrecisionCascades = (!precisionCascades.empty()); // decide before inserting primary buffer pointer at beginning
+      if (doPrecisionCascades) {
+         precisionCascades.insert(precisionCascades.begin(),bufcur);
```

Comment:
Instead of doing an insert, add at line 1297:
```
precisionCascades.push_back(bufcur);
```

Reply:
OK, I can do that. The only thing to take into consideration is the case of the precision cascade where there are no auxiliaries defined, just the primary. I believe in that case it's arbitrary whether to treat it as a precision cascade or just a normal single buffer, but it should be fine to do so. I'll change it.

Reply:
>  or just a normal single buffer, but it should be fine to do so. I'll change it.

As far as I could tell, the code (here) was already doing so.  I believe that treating the no-auxialiary as a normal single buffer is (slightly) more efficient as it avoids the passing/checking of the arrays/vectors.